website/content/docs/api-gateway/configuration/gateway.mdx

```diff
@@ -0,0 +1,325 @@
+|`values` | Specifies an array of string values. If `operator` is configured to `In` or `NotIn`, then the `values` array must contain values. If `operator` is configured to `Exists` or `DoesNotExist`, then the `values` array must be empty. This array is replaced during a strategic merge patch. | array of strings | required when `matchExpressions` is declared |
+
+In the following example, routes in namespaces that contain `foo` and `bar` are allowed to attach routes to the `Gateway`. 
+```yaml
+namespaceSelector:
+  matchExpressions:
+  - key: kubernetes.io/metadata.name
+    operator: In
+    values:
+    - foo
+    - bar
+```
+
+### listeners.allowedRoutes.namespaces.selector.matchLabels
+Specifies an array of labels and label values. If a match is found, then routes with the matching label(s) are allowed to attach to the `Gateway`. This selector can contain any arbitrary key/value pair.
+
+In the following example, routes in namespaces that have a `bar` label are allowed to attach to the `Gateway`.
+
+```yaml
+namespaceSelector:
+    matchLabels:
+      foo: bar
+```
+
+Refer to [Labels and Selectors](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) in the Kubernetes documentation for additional information about labels.
+
+### listeners.hostname
+Specifies the `listener`'s hostname.
+* Type: string
+* Required: required
+
+### listeners.name
+Specifies the `listener`'s name.
+* Type: string
+* Required: required
+
+### listeners.port
+Specifies the port number that the `listener` attaches to.
+* Type: integer
+* Required: required
+
+### listeners.protocol
+Specifies the protocol the `listener` communicates on.
+* Type: string
+* Required: required
+
+Allowed values are `TCP`, `HTTP`, or `HTTPS`
+
+### listeners.tls
+Specifies the `tls` configurations for the `Gateway`. The `tls` object is required if `protocol` is set to `HTTPS`. The object contains the following fields:
+
+| Parameter | Description | Type | Required |
+| ---       | ---         | ---  | ---      |
+| `certificateRefs` | <div style={{width:480}}>Specifies Kubernetes `name` and `namespace` objects that contains TLS certificates and private keys. <br/>The certificates establish a TLS handshake for requests that match the `hostname` of the associated `listener`. Each reference must be a Kubernetes Secret. If you are using a Secret in a namespace other than the `Gateway`'s, each reference must also have a corresponding [`ReferencePolicy`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferencePolicy).</div>  | Object or array | Required if `tls` is set |
+| `mode` | Specifies the ? | string | Required if `certificateRefs` is set |
+| `options` | ??? | Map of strings | optional |
+
+In the following example, `tls` settings are configured . . .
+
+```yaml
+
+tls:
+  certificateRefs: 
+    name: ?
+    namespace: ?
+  mode: ?
+  options: 
+  - ?
+  - ? 
+  - ?
```

Comment:
```suggestion
| `mode` | Specifies the ? | string | Required if `certificateRefs` is set |
| `options` | ??? | Map of strings | optional |

In the following example, `tls` settings are configured . . .

```yaml

tls:
  certificateRefs: 
    name: ?
    namespace: ?
  mode: ?
  options: 
  - ?
  - ? 
  - ?
```
Flagging this as a section that needs some help. I don't know what the `mode` option is or what the values are.

Reply:
I had to reverse engineer some code, but I added some additional context. I would raise TLS Options as a section that could do with an additional PR focused solely on improving the clarity of those settings.